### DISCUSSION TITLE
TypeError: p5.Vector is undefined

### DISCUSSION THREAD

**Discussion by hanhanxue** [P5-wrapper/react#382] [Q&A] [answered]
Hello!

I'm trying to create a vector using `p5.Vector.fromAngle()` in nextjs and it's throwing an error that it's undefined, but `p5.createVector()` works

**Comment by jamesrweb** [comment, marked as answer]
This is covered in the documentation under the "Constructors" section. Why do you need the constructor though when you can just `createVector`?

**Reply by hanhanxue** [reply]
Got it! I didn't know it was already covered :) I'm trying to use the `fromAngle()` method of Vector.

@jamesrweb again thank you so much for the replies it's been really helpful!

**Reply by jamesrweb** [reply]
Always happy to help, @hanhanxue ☺️

**Reply by hanhanxue** [reply]
@jamesrweb so somewhat related, how do I define Vector type if I want to have something like this:

```ts
class Particle {
  pos: p5.Vector
  vel: p5.Vector
  acc: p5.Vector

  constructor(p5: P5CanvasInstance) {
    this.pos = p5.createVector(0, 0)
    this.vel = p5.createVector(0, 0)
    this.acc = p5.createVector(0, 0)
  }

  update(): void {
    this.vel.add(this.acc)
    this.pos.add(this.vel)
  }
}
```

Obviously as is the p5 namespace is missing

**Reply by jamesrweb** [reply]
Just add `import p5 from "p5";` to your imports list.

If you need `p5` as a type in and of itself, use `import type p5 from "p5";` instead.

Some example usages of each of these options can be found [here for the type import](https://github.com/P5-wrapper/react/blob/1b7828b7b0a609f659002e501febe3fa53f27805/src/contracts/P5CanvasInstance.ts) and [here for the value import](https://github.com/P5-wrapper/react/blob/1b7828b7b0a609f659002e501febe3fa53f27805/src/utils/createCanvasInstance.ts).
